src/source_clang.cc

```diff
@@ -859,20 +860,22 @@ Source::ClangViewAutocomplete(file_path, language) {
   });
   
   get_token=[this]() -> Token {
-    if(parsed) {
-      auto iter=get_buffer()->get_insert()->get_iter();
-      auto line=(unsigned)iter.get_line();
-      auto index=(unsigned)iter.get_line_index();
-      for(auto &token: *clang_tokens) {
-        auto cursor=token.get_cursor();
-        if(token.get_kind()==clang::Token_Identifier && cursor.has_type()) {
-          if(line==token.offsets.first.line-1 && index>=token.offsets.first.index-1 && index <=token.offsets.second.index-1) {
-            if(static_cast<unsigned>(token.get_cursor().get_kind())==103) //These cursors are buggy
-              continue;
-            auto referenced=cursor.get_referenced();
-            if(referenced)
-              return Token(this->language, static_cast<int>(referenced.get_kind()), token.get_spelling(), referenced.get_usr());
-          }
+    if(!parsed) {
+      Info::get().print("Buffer is parsing");
```

Comment:
```cpp
if(!parsed) {
  Info::get().print("Buffer is parsing");
  return location
}
// could be something like
if(Info::get().assert(parsed, "Buffer is parsing"))
  return location;
```
I only saw this because it seems a condition is checked before every print and we get a lot of duplication. I realise it might be to much abstraction when thinking of readability :P